src/hooks/useDebounce.ts

```diff
@@ -0,0 +1,28 @@
+/* eslint-disable react-hooks/exhaustive-deps */
+import debounce from 'lodash.debounce';
+import { useCallback, useEffect, useRef } from 'react';
+
+type DefaultCallbackArgs = Array<unknown>;
+type CallbackFn<Args extends DefaultCallbackArgs, ReturnType> = (
+    ...args: Args
+) => Promise<ReturnType>;
+
+
+/**
+ * Runs the callback function only if the delay time has passed.
+ */
+export default function useDebounce<Args extends DefaultCallbackArgs, ReturnType>(
+    callback: CallbackFn<Args, ReturnType>,
+    delay: number,
+): (...args: Args) => Promise<ReturnType> | undefined {
```

Comment:
Why would this function return `undefined`? (Also see comment at call site in `index.tsx`).

Reply:
As mentioned, this has to do with the output of a debounced function from lodash debounce.

Reply:
Aah was fooled by the `|` (or rather my misunderstanding of the order in which the operands of `|` are interpreted).